flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -303,6 +348,11 @@ protected MainThreadExecutor getMainThreadExecutor() {
         return mainThreadExecutor;
     }
 
+    @VisibleForTesting
+    CloseableRegistry getResourceRegistry() {
+        return resourceRegistry;
+    }
```

Comment:
I think the tests should not require exposing the closeable registry. Instead, they should validate that the executors are properly closed.